python/ember/input.py

```diff
@@ -1006,23 +1029,28 @@ def run(self, command=None):
         If the script which calls this function is passed the argument
         *validate*, then the configuration will be checked for errors and the
         script will exit without running the simulation.
-        """
-        if len(sys.argv) > 1 and sys.argv[1].lower() == 'validate':
-            # Validate the configuration and exit
-            self.validate()
-            return
 
-        if command is None:
-            self.validate()
-        elif command != 'force':
-            print('An argument of "force" will allow for skipping validation and attempting to simulate.')
-            print('Exiting...')
+        If the script which calls this function is passed the argument
+        *extinction*, then the strain rate parameter will be increased until
+        a steady burning flame can no longer be achieved.
+        """
 
         concrete = self.evaluate()
-        if self.strainParameters.rates:
-            return concrete.multirun()
+
+        if command == 'validate':
+            self.validate()
+            return
+        elif command == 'extinction':
+            return concrete.runESR()
+        elif command == None:
+            if self.strainParameters.rates:
+                return concrete.multirun()
+            else:
+                return concrete.run()
```

Comment:
This breaks the old capability of calling the validator as a standalone action and skipping the actual 'run', e.g. by running
```
$ python myscript.py validate
```
from the command line.

Maybe it would make more sense to have a `runESR` method on this class as well, rather than repurposing the existing flags.

Reply:
We could go that direction, but isn't it simple to bring back this capability by bringing back the lines:
'''
        if len(sys.argv) > 1 and sys.argv[1].lower() == 'validate':
            # Validate the configuration and exit
            self.validate()
            return
'''
?

Reply:
We could, but you're still changing the meaning of the argument to the `run` function. In addition, `validate` or `force` seem like they should be independent of whether or not you're doing an ESR calculation, so they shouldn't be activated the same way. I think the simplest way to separate these concerns is to have the function that you call to run be different. Unless you want it to be contingent on a different option that is part of the `config`.

Reply:
(which would make this more similar to the way that we dispatch to `concrete.run` vs `concrete.multirun`)